### PR TITLE
Replace flake8 and isort with ruff

### PR DESCRIPTION
**Pull request by ticosax**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,7 +4,7 @@ envlist =
     py{37,38,39,310}-dj-3.2
     py{38,39,310}-dj-4.0
     py{310}-dj-master
-    flake8
+    ruff
```
Comment (MRigal) [inline tox.ini:7]: This is I think not at all in use by tox, or am I wrong? I think we would need an additional job and also additional config in GH actions, invoking `pre-commit` see example here https://github.com/ansible/ansible-lint/blob/main/tox.ini#L74 (probably a separated PR)

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -33,22 +33,8 @@ repos:
           - "--line-length"
           - "119"
         exclude: ^django_fsm_log/migrations|^docs/
-  - repo: https://github.com/pycqa/isort
-    rev: 5.12.0
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    # Ruff version.
+    rev: 'v0.0.257'
```
Comment (MRigal) [inline .pre-commit-config.yaml:38]: Just from the order, https://github.com/charliermarsh/ruff-pre-commit says it should be run before other things like black

Comment (MRigal) [reply .pre-commit-config.yaml:38]: You can also remove yesqa

**Review by MRigal — CHANGES_REQUESTED** [review]
LGTM generally, added some feedback, maybe one point to be put in a separate issue/PR

File: tox.ini (hunk 1 above)
Comment (ticosax) [reply tox.ini:7]: good point, thanks for the pointers.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (ticosax) [reply .pre-commit-config.yaml:38]: thanks

Comment (ticosax) [reply .pre-commit-config.yaml]: I also removed pyupgrade as it is also covered by ruff

**Review by MRigal — APPROVED** [review]
Great job! Should we add an issue to add again a (working) ruff check job on Github Actions?

**Comment by ticosax** [comment]
> Great job! Should we add an issue to add again a (working) ruff check job on Github Actions?

I will have time to make a PR right away

File: tox.ini (hunk 1 above)
Comment (MRigal) [reply tox.ini]: Actually, we run pre-commit.ci and it is a blocking check on GitHub, so no need to add anything here, it was just requiring to be removed here :-)

**Pull request merged**